### PR TITLE
Improving Code readability of Tests

### LINKED ISSUE #75: Add documentation for Testing Framework
We are using pytest for testing and tox to automate this process. 

While we have set-up the scripts, we need to add documentation on how to write Unit Tests as well as establish standards to write these tests.

### PR DESCRIPTION
**Pull request by NileshPant1999**
## Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes #75 

## Type of change

Please mark options that are relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## How Has This Been Tested?

Did you run `make test-all`?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

## Checklist:

- [ ] New Unit tests added
- [x] Unit tests pass locally with my changes

### CODE REVIEW

File: tests/base/test_logging.py [hunk 1]
```diff
@@ -1,34 +1,42 @@
...
 import pydp as pd
-
+# Creating a Class TestLogging
 class TestLogging():
+    #Creating First testcase
     def test_logging(self):
         directory = 'tests'
         vlog_level = 3
         l = pd.logging.init_logging(directory, 'log_test', vlog_level)
+        #if both directory and vlog_level contains value the test will be pass
```
Comment (chinmayshah99) [inline tests/base/test_logging.py]: This comment is misleading. 
`pd.logging.init_logging`  returns true if it's successful in initializing logs. Check for DP code: https://github.com/google/differential-privacy/blob/master/differential_privacy/base/logging.h#L289 for more.

File: tests/base/test_logging.py [hunk 2]
```diff
@@ -1,34 +1,42 @@
...
         vlog_level = 3
         l = pd.logging.init_logging(directory, 'log_test.txt', vlog_level)
+        #location of directory must be given otherwise test will fail
         assert l == False
 
     def test_logging2(self):
         directory = 'tests'
         vlog_level = 3
         l = pd.logging.init_logging(directory, '', vlog_level)
+        #All the attributes of init_logging must be given otherwise test will fail
```
Comment (chinmayshah99) [inline tests/base/test_logging.py]: A proper description of the tests needs to be given. 
This test fails if the output file is not mentioned.

File: tests/base/test_logging.py [hunk 3]
```diff
@@ -1,34 +1,42 @@
...
         assert l == False
 
+    # Creating an instance which will return level of vlog
```
Comment (chinmayshah99) [inline tests/base/test_logging.py]: This comment does not describe the test. This comment can be either removed or L#33 text must be mentioned here.

File: tests/base/test_logging.py [hunk 4]
```diff
@@ -1,34 +1,42 @@
...
     def test_vlog_level(self):
         directory = 'tests'
         vlog_level = 1
         l = pd.logging.init_logging(directory, 'log_test.txt', vlog_level)
+        #outcome must be equal to vlog_level otherwise the code will fail
         assert pd.logging.get_vlog_level() == vlog_level
-    
+
+    # Creating an Instance which will return the test_vlog_directory
```
Comment (chinmayshah99) [inline tests/base/test_logging.py]: This comment does not describe the test. This comment can be either removed or L#41 text must be mentioned here.

File: tests/base/test_status.py [hunk 5]
```diff
@@ -29,9 +29,9 @@ def test_status_codes(self):
             "StatusCode.kUnauthenticated", 
         ]
         actual_codes = []
-        x = range(17)
-        for n in x:
-            actual_codes.append(str(pd.Status.StatusCode(n)))
+        
+        
+        [actual_codes.append(str(pd.Status.StatusCode(n))) for n in range(17)]
```
Comment (chinmayshah99) [inline tests/base/test_status.py]: Is there any reason for using generator?
If I print `actual_codes ` after running`[actual_codes.append(str(pd.Status.StatusCode(n))) for n in range(17)]`, I get a generator object. Just curious does it give any advantage?

Comment (jeamick) [reply tests/base/test_status.py:34]: Not really, we performance don't matter here until it will ...
So, for now, it should be fine.

Comment (chinmayshah99) [reply tests/base/test_status.py:34]: Then let's discard this change. It reduces the readability of the code.

Comment (jeamick) [reply tests/base/test_status.py:34]: yup

**Comment by chinmayshah99** [comment]
Inactive

**Pull request closed without merging**